### ISSUE TITLE
Elisp checker doesn't identify variable in macro

### ISSUE THREAD

**Issue by jschaf** [flycheck/flycheck#818]
At `(my-macro)` flycheck gives the following error:

`Symbol's value as variable is void: my-var`

The byte-compiler does not show an error.

```emacs-lisp
;;; bug.el --- bug

;;; Commentary:
;;

;;; Code:

(defvar my-var nil)

(defmacro my-macro ()
  "Macro."
  my-var)

(my-macro)


(provide 'bug)
;;; bug.el ends here
```

**Comment by cpitclaudel** [comment]
This is not a Flycheck bug: `byte-compile-file` produces the same output. Or did I misunderstand something?

**Comment by jschaf** [comment]
Oh, must have had it defined in my environment.  No bug.
